benchmarks/src/main/java/io/druid/benchmark/query/TimeseriesBenchmark.java

```diff
@@ -0,0 +1,349 @@
+          {
+            return 0;
+          }
+        }
+    );
+    INDEX_MERGER_V9 = new IndexMergerV9(JSON_MAPPER, INDEX_IO);
+  }
+
+  private static final Map<String, Map<String, TimeseriesQuery>> SCHEMA_QUERY_MAP = new LinkedHashMap<>();
+  static {
+    // queries for the basic schema
+    Map<String, TimeseriesQuery> basicQueries = new LinkedHashMap<>();
+    BenchmarkSchemaInfo basicSchema = BenchmarkSchemas.SCHEMA_MAP.get("basic");
+
+    { // basic.A
+      QuerySegmentSpec intervalSpec = new MultipleIntervalSegmentSpec(Arrays.asList(basicSchema.getDataInterval()));
+
+      List<AggregatorFactory> queryAggs = new ArrayList<>();
+      queryAggs.add(new LongSumAggregatorFactory("sumA", "sumA"));
+      queryAggs.add(new LongMaxAggregatorFactory("maxB", "maxB"));
+      queryAggs.add(new DoubleSumAggregatorFactory("sumC", "sumC"));
+      queryAggs.add(new DoubleMinAggregatorFactory("minD", "minD"));
+      queryAggs.add(new HyperUniquesAggregatorFactory("hyperUniquesMet", "hyper"));
+
+      TimeseriesQuery queryA =
+          Druids.newTimeseriesQueryBuilder()
+                .dataSource("blah")
+                .granularity(QueryGranularities.ALL)
+                .intervals(intervalSpec)
+                .aggregators(queryAggs)
+                .descending(false)
+                .build();
+
+      basicQueries.put("A", queryA);
+    }
+
+    SCHEMA_QUERY_MAP.put("basic", basicQueries);
+  }
+
+  @Setup
+  public void setup() throws IOException
+  {
+    System.out.println("SETUP CALLED AT " + System.currentTimeMillis());
+
+    if (ComplexMetrics.getSerdeForType("hyperUnique") == null) {
+      ComplexMetrics.registerSerde("hyperUnique", new HyperUniquesSerde(Hashing.murmur3_128()));
+    }
+
+    executorService = Execs.multiThreaded(numSegments, "TimeseriesThreadPool");
+
+    String[] schemaQuery = schemaAndQuery.split("\\.");
+    String schemaName = schemaQuery[0];
+    String queryName = schemaQuery[1];
+
+    schemaInfo = BenchmarkSchemas.SCHEMA_MAP.get(schemaName);
+    query = SCHEMA_QUERY_MAP.get(schemaName).get(queryName);
+
+    incIndexes = new ArrayList<>();
+    for (int i = 0; i < numSegments; i++) {
+      System.out.println("Generating rows for segment " + i);
+      BenchmarkDataGenerator gen = new BenchmarkDataGenerator(
+          schemaInfo.getColumnSchemas(),
+          RNG_SEED + i,
+          schemaInfo.getDataInterval(),
+          rowsPerSegment
+      );
+
+      IncrementalIndex incIndex = makeIncIndex();
+
+      for (int j = 0; j < rowsPerSegment; j++) {
+        InputRow row = gen.nextRow();
+        if (j % 10000 == 0) {
+          System.out.println(j + " rows generated.");
+        }
+        incIndex.add(row);
+      }
+      System.out.println(rowsPerSegment + " rows generated");
+      incIndexes.add(incIndex);
+    }
+
+    File tmpFile = File.createTempFile("TopNBenchmark-" + System.currentTimeMillis(), ".TEMPDIR");
+    tmpFile.delete();
+    tmpFile.mkdirs();
```

Comment:
`Files.createTempDir` does all this for you without introducing a possible race condition.